corelib/src/LocalGridMaker.cpp

```diff
@@ -258,26 +258,7 @@ void LocalGridMaker::createLocalMap(
 
 				UDEBUG("scan format=%d", scan.format());
 
-				bool normalSegmentationTmp = normalsSegmentation_;
-				float minGroundHeightTmp = minGroundHeight_;
-				float maxGroundHeightTmp = maxGroundHeight_;
-				if(scan.is2d())
-				{
-					// if 2D, assume the whole scan is obstacle
-					normalsSegmentation_ = false;
-					minGroundHeight_ = std::numeric_limits<int>::min();
-					maxGroundHeight_ = std::numeric_limits<int>::min()+100;
-				}
-
 				createLocalMap(scan, node.getPose(), groundCells, obstacleCells, emptyCells, viewPoint);
-
-				if(scan.is2d())
-				{
-					// restore
-					normalsSegmentation_ = normalSegmentationTmp;
-					minGroundHeight_ = minGroundHeightTmp;
-					maxGroundHeight_ = maxGroundHeightTmp;
-				}
```

Comment:
This is done in a special case when we combine 2D lidar data and 3D depth data into the local grid. Is it really never called even in that case?

Reply:
It grabbed my attention because of multiple checks for is2d() and occupancy == 0.  However, after a second look, you are correct that occupancy == 2 could still get there.    

My bad.